src/zenml/integrations/kserve/model_deployers/kserve_model_deployer.py

```diff
@@ -0,0 +1,464 @@
+            "PREDICTION_URL": service_instance.prediction_url,
+            "MODEL_URI": service_instance.config.model_uri,
+            "MODEL_NAME": service_instance.config.model_name,
+            "KSERVE_INFERENCE_SERVICE": service_instance.crd_name,
+        }
+
+    @staticmethod
+    def get_active_model_deployer() -> "KServeModelDeployer":
+        """Get the KServe model deployer registered in the active stack.
+
+        Returns:
+            The KServe model deployer registered in the active stack.
+
+        Raises:
+            TypeError: if the KServe model deployer is not available.
+        """
+        model_deployer = Repository(  # type: ignore [call-arg]
+            skip_repository_check=True
+        ).active_stack.model_deployer
+        if not model_deployer or not isinstance(
+            model_deployer, KServeModelDeployer
+        ):
+            raise TypeError(
+                f"The active stack needs to have a KServe model deployer "
+                f"component registered to be able to deploy models with KServe "
+                f"You can create a new stack with a KServe model "
+                f"deployer component or update your existing stack to add this "
+                f"component, e.g.:\n\n"
+                f"  'zenml model-deployer register kserve --flavor={KSERVE_MODEL_DEPLOYER_FLAVOR} "
+                f"--kubernetes_context=context-name --kubernetes_namespace="
+                f"namespace-name --base_url=https://ingress.cluster.kubernetes'\n"
+                f"  'zenml stack create stack-name -d kserve ...'\n"
+            )
+        return model_deployer
+
+    @property
+    def kserve_client(self) -> KServeClient:
+        """Get the KServe client associated with this model deployer.
+
+        Returns:
+            The KServeclient.
+        """
+        if not self._client:
+            self._client = KServeClient(
+                context=self.kubernetes_context,
+            )
+        return self._client
+
+    # TODO[High]: Save ZenML Secrets in a temp file and load them into
+    # the KServe credentials
+    def _set_credentials(self) -> None:
+        """Set the credentials for the given service instance.
+
+        Raises:
+            RuntimeError: if the credentials are not available.
+            ValueError: if the credentials are not valid.
+        """
+        if self.secret:
+            secret_manager = Repository(  # type: ignore [call-arg]
+                skip_repository_check=True
+            ).active_stack.secrets_manager
+
+            if not secret_manager or not isinstance(
+                secret_manager, BaseSecretsManager
+            ):
+                raise RuntimeError(
+                    f"The active stack doesn't have a secret manager component. "
+                    f"The ZenML secret specified in the KServe Model "
+                    f"Deployer configuration cannot be fetched: {self.secret}."
+                )
+            if self.kserve_client:
```

Comment:
Do we need this `if` here, seems to me like that property will always have a value?